src/common/bitstream.cpp

```diff
@@ -0,0 +1,392 @@
+#include <algorithm>
+#include "bitstream.h"
+#include "util.h"
+
+#define BITS_TO_BYTES(bits) (((bits) + 7) / 8)
+
+
+BitStream::BitStream(std::vector<uint8_t>& exisitingBuf) :
+    buf(exisitingBuf),
+    streamBitPos(0),
+    lastError(Error::NONE) {
+
+}
+
+size_t BitStream::getSizeBits() const {
+    return buf.size() * 8;
+}
+
+size_t BitStream::getRemainingBits() const {
+    size_t sizeBits = getSizeBits();
+    if(streamBitPos >= sizeBits) {
```

Comment:
Correcting, in a big way.

The VS auto-formatter was being a pain for me and I felt it had way too many settings to want to fiddle with, so I admit to just having turned the whole thing off at some earlier point. =P  I can't avoid that anymore.  (The said setting was off for me, in any case, so even if ther auto-formatter had previously been "on" ...)